spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/StoredProcOutboundGateway.java

```diff
@@ -56,6 +87,14 @@ public String getComponentType() {
 	}
 
 	@Override
+	protected void doInit() {
+		super.doInit();
+		if (!this.requiresReplyExplicitlySet) {
+			setRequiresReply(this.expectSingleResult);
+		}
```

Comment:
Well, there is no guaranty in the order how setters are called via XML parser.
So, we may end up with undesired affect, when `setRequiresReply()` is called later with `false`, although we would like true by the `expectSingleResult`.

A similar issue with non-idempotent behavior with raw java, when we change the order how we call setters.

With a `requiresReplyExplicitlySet` flag and modification in the `doInit()` we reach a robust state.

Am I missing something ?